contracts/ServiceRegistryTokenUtility.sol

```diff
@@ -148,28 +155,35 @@ contract ServiceRegistryTokenUtility is IErrorsRegistries {
     }
 
     /// @dev Creates a record with the token-related information for the specified service.
+    /// @notice We assume that the token is checked for being a non-zero address and a non-ETH address representation
+    ///         outside of this function. Here we optimistically check for the token to have a specific `balanceOf`
+    ///         view function. It is possible this is the attacker token that has all the required functions defined
+    ///         correctly, so there is no point in checking that formality. All the required checks will be done in-place
+    ///         where the possibility of misbehavior can be caught by return values of token function.
     /// @param serviceId Service Id.
     /// @param token Token address.
     /// @param agentIds Set of agent Ids.
     /// @param bonds Set of correspondent bonds.
-    function createWithToken(uint256 serviceId, address token, uint32[] memory agentIds, uint256[] memory bonds) external {
+    function createWithToken(
+        uint256 serviceId,
+        address token,
+        uint32[] memory agentIds,
+        uint256[] memory bonds
+    ) external
+    {
         // Check for the manager privilege for a service management
         if (manager != msg.sender) {
             revert ManagerOnly(msg.sender, manager);
         }
 
-        // Check for the zero token address
-        if (token == address(0)) {
-            revert ZeroAddress();
-        }
         // TODO Check for the token ERC20 formality
         
         uint256 securityDeposit;
         // Service is newly created and all the array lengths are checked by the original ServiceRegistry create() function
         for (uint256 i = 0; i < agentIds.length; ++i) {
-            // Check for a non-zero bond value
+            // Check for a non-zero bond value and skip those with zeros (possible when updating a service)
```

Comment:
when this function is used to `update` the service, we can have some bonds equal to zero. But we still need to check that there is at least one i such that bond[i] is not zero, correct?

Reply:
Ah no, I see from the next PR that a check is made at the serviceManager level. Correct?

Reply:
yes